core/src/main/scala/org/apache/spark/deploy/worker/ExecutorRunner.scala

```diff
@@ -206,11 +206,11 @@ private[deploy] class ExecutorRunner(
       case interrupted: InterruptedException =>
         logInfo("Runner thread for executor " + fullId + " interrupted")
         state = ExecutorState.KILLED
-        killProcess(None)
+        killProcess(s"Runner thread for executor $fullId interrupted")
       case e: Exception =>
         logError("Error running executor", e)
         state = ExecutorState.FAILED
-        killProcess(Some(e.toString))
+        killProcess(s"Error running executor: $e")
```

Comment:
Could you dump the full stacktrace for the error message?

Reply:
The message will be sent to Master and Driver. Do you think including the full stacktrace might make it too large? How about logging it in the Worker log here?

Reply:
> logError("Error running executor", e)

This ^ should already log the full stack in the worker log.

My concern is that the exception that throw here could be rare case and full stack will help the debug. Since worker already logs it, I'm fine to return a short error message to master.